examples/conanfile/layout/editable_components.rst

```diff
@@ -0,0 +1,123 @@
+.. _examples_conanfile_layout_components_editables:
+
+Using components and editable packages
+--------------------------------------
+
+It is possible to define components in the ``layout()`` method, to support the case of ``editable`` packages.
+That is, if we want to put a package in ``editable`` mode, and that package defines ``components``, it is 
+necessary to define the components layout correctly in the ``layout()`` method.  
+Let's see it in a real example.
+
+Please, first of all, clone the sources to recreate this project. You can find them in the
+`examples2.0 repository <https://github.com/conan-io/examples2>`_ in GitHub:
+
+.. code-block:: bash
+
+    $ git clone https://github.com/conan-io/examples2.git
+    $ cd examples2/examples/conanfile/layout/editable_components
+
+There we find a ``greetings`` subfolder and package, that contains 2 libraries, the ``hello`` library and the
+``bye`` library. Each one is modeled as a ``component`` inside the package recipe:
+
+.. code-block:: python
+    :caption: greetings/conanfile.py
+
+    class GreetingsConan(ConanFile):
+        name = "greetings"
+        version = "0.1"
+        settings = "os", "compiler", "build_type", "arch"
+        generators = "CMakeDeps", "CMakeToolchain"
+        exports_sources = "src/*"
+
+        def build(self):
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
+
+        def layout(self):
+            cmake_layout(self, src_folder="src")
+            # This "includedirs" starts in the source folder, wich is "src"
+            # So the components include dirs is the "src" folder (includes are
+            # intended to be included as ``#include "hello/hello.h"``)
+            self.cpp.source.components["hello"].includedirs = ["."]
```

Comment:
do I have to write that for all the components, or can I put only particular components into editable mode?

Reply:
The package is in editable mode means that all of its components are in editable mode. Recall that components are not a packaging concept or a distribution concept, they cannot be handled separately, only linked separately.